### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by vectrixdevelops**
Work in progress PR to cleanup Guardian.

- [x] Simplify and Enhance Stability for the Context Registry / Context System
- [x] General Style Fixes
- [x] Minimize Check Logic (and improve efficiency)
- [x] Shaded Jar Size Reduction

### CODE REVIEW

File: src/main/java/io/github/connorhartley/guardian/internal/checks/HorizontalSpeedCheck.java [hunk 1]
```diff
@@ -111,120 +108,145 @@ public SequenceBlueprint getSequence() {
...
+                                                    " while they're " + playerControlState.name() + ".");
+
+                                    if (travelDisplacement > maximumSpeed) {
+                                        report
+                                                .information("Overshot maximum speed by " + (travelDisplacement - maximumSpeed) + ".")
+                                                .type("horizontally overspeeding")
+                                                .severity(travelDisplacement - maximumSpeed);
+
+                                        // TODO : Remove this after testing \/
+                                        plugin.getLogger().warn(user.getName() + " has triggered the horizontal speed check and overshot " +
```
Comment (parlough) [inline src/main/java/io/github/connorhartley/guardian/internal/checks/HorizontalSpeedCheck.java:241]: You utilize logging levels. Why not have a debug level so you don't have to remind yourself to remove these as they could be good to keep for future debugging purposes.

File: src/main/java/io/github/connorhartley/guardian/internal/detections/FlyDetection.java [hunk 2]
```diff
@@ -161,139 +168,229 @@ public Guardian getPlugin() {
...
+                    this.configTickBounds.<ConfigurationNode>loadStorage(this.configurationNode);
+                    this.configPunishmentLevels.<ConfigurationNode>loadStorage(this.configurationNode);
+                    this.configPunishmentProperties.<ConfigurationNode>loadStorage(this.configurationNode);
+                    this.configCustomPunishments.<ConfigurationNode>loadStorage(this.configurationNode);
+                    this.configSeverityDistribution.<ConfigurationNode>loadStorage(this.configurationNode);
+
+                    this.configManager.save(this.configurationNode);
+                }
+            } catch (IOException e) {
+                this.flyDetection.getPlugin().getLogger().error("A problem occurred attempting to load Guardians global configuration!", e);
```
Comment (parlough) [inline src/main/java/io/github/connorhartley/guardian/internal/detections/FlyDetection.java]: Guardian's* :P

Comment (parlough) [reply src/main/java/io/github/connorhartley/guardian/internal/detections/FlyDetection.java]: Same thing on line 325. Also you forget it in a lot of the other config log statements actually.

File: src/main/java/io/github/connorhartley/guardian/internal/punishments/CustomPunishment.java [hunk 3]
```diff
@@ -23,19 +23,18 @@
...
+import io.github.connorhartley.guardian.storage.container.StorageKey;
 import io.github.connorhartley.guardian.storage.container.StorageValue;
 import org.spongepowered.api.Sponge;
 import org.spongepowered.api.entity.living.player.User;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Optional;
+import java.util.*;
```
Comment (parlough) [inline src/main/java/io/github/connorhartley/guardian/internal/punishments/CustomPunishment.java]: You're not using that many, no need to use star import.

File: src/main/java/io/github/connorhartley/guardian/sequence/SequenceController.java [hunk 4]
```diff
@@ -78,7 +78,12 @@ public void invoke(Player player, Event event) {
                 return false;
             }
 
-            SequenceFinishEvent attempt = new SequenceFinishEvent(sequence, player, sequence.getSequenceReport(), Cause.of(NamedCause.source(this.plugin), NamedCause.of("CONTEXT", sequence.getContext())));
+            if (sequence.hasStarted()) {
+                sequence.getContextHandler().stop();
+            }
+
+            SequenceFinishEvent attempt = new SequenceFinishEvent(sequence, player, sequence.getSequenceReport(),
+                    Cause.of(NamedCause.source(this.plugin), NamedCause.of("CONTEXT", sequence.getContextContainer())));
```
Comment (parlough) [inline src/main/java/io/github/connorhartley/guardian/sequence/SequenceController.java:86]: I'm curious if Guardian should create its own cause to use for these sequence events or is that unnecessary?

File: src/main/java/io/github/connorhartley/guardian/internal/detections/SpeedDetection.java [hunk 5]
```diff
@@ -160,161 +167,257 @@ public Guardian getPlugin() {
...
+                controlValues.put("fly", 1.075);
+
+                this.configControlValues = new StorageValue<>(new StorageKey<>("control-values"),
+                        "Magic values for movement the player controls that are added each tick.",
+                        controlValues, new TypeToken<Map<String, Double>>() {
+                });
+
+                // Block Speed
+
+                HashMap<String, Double> materialValues = new HashMap<>();
```
Comment (parlough) [inline src/main/java/io/github/connorhartley/guardian/internal/detections/SpeedDetection.java]: Any need to keep all of the Maps in the class as HashMap<>? If not, might as well switch to Map<>.

File: src/main/java/io/github/connorhartley/guardian/internal/detections/FlyDetection.java (hunk 2 above)
Comment (vectrixdevelops) [reply src/main/java/io/github/connorhartley/guardian/internal/detections/FlyDetection.java]: Also the fact that it is talking about the "global configuration" when it is only working with the detection's configuration.

File: src/main/java/io/github/connorhartley/guardian/internal/detections/FlyDetection.java [hunk 6]
```diff
@@ -161,139 +168,229 @@ public Guardian getPlugin() {
...
+//              punishmentLevels.put("flag", 0.2);
+//              punishmentLevels.put("report", 0.3);
+//              punishmentLevels.put("kick", 0.5);
+
+                this.configPunishmentLevels = new StorageValue<>(new StorageKey<>("punishment-levels"),
+                        "Punishments that happen when the user reaches the individual severity threshold.",
+                        punishmentLevels, new TypeToken<Map<String, Double>>() {
+                });
+
+                HashMap<String, String> punishmentProperties = new HashMap<>();
```
Comment (parlough) [inline src/main/java/io/github/connorhartley/guardian/internal/detections/FlyDetection.java]: Map thing in this class as well.

**Comment by parlough** [comment]
Looks good to me. I left some small comments related to style things, but yeah good job. This is a good step towards being easier to contribute to the project and simply being more unified throughout while also fixing some issues :)

Guardian will be joining you soon. 💂 💂‍♀️ 🛡

File: src/main/java/io/github/connorhartley/guardian/internal/detections/FlyDetection.java (hunk 2 above)
Comment (parlough) [reply src/main/java/io/github/connorhartley/guardian/internal/detections/FlyDetection.java]: Make sure to check the other configuration's log statements as well then :)

**Comment by parlough** [comment]
LGTM - Merge it in
unnamed

**Pull request merged**